ext/winevt/winevt_channel.c

```diff
@@ -254,8 +264,33 @@ rb_winevt_channel_each(VALUE self)
     }
 
     status = is_subscribable_channel_p(hChannelConfig, winevtChannel->force_enumerate);
-    if (status != ERROR_SUCCESS)
+    EvtClose(hChannelConfig);
+
+    if (status == ERROR_OUTOFMEMORY) {
+      EvtClose(winevtChannel->channels);
+      winevtChannel->channels = NULL;
+
+      free(buffer);
+      buffer = NULL;
+      bufferSize = 0;
+
+      rb_raise(rb_eRuntimeError, "realloc failed\n");
+    } else if (status == ERROR_INVALID_DATA) {
+      free(buffer);
```

Comment:
Since this block isn't error, it should be a separated statement to clarify it.

Reply:
Right. Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/23/commits/99673b6e617854afd8d6b0c9dd8756f6349e055a.